include/sys/util.h

```diff
@@ -25,6 +25,9 @@
 #define POINTER_TO_INT(x)  ((intptr_t) (x))
 #define INT_TO_POINTER(x)  ((void *) (intptr_t) (x))
 
+/* Helper to get the difference between two pointers as an int */
+#define PTR_DIFF(y, x) ((int)((intptr_t)(y) - (intptr_t)(x)))
```

Comment:
> large enough in practice.

How about "PTR_TO_INT_DIFF(x, y)" instead? This would consistent with the other macros above and  it would not miss the important "_INT_" part that guards the user against using it for a distance bigger than 2^sizeof(int).

Now it would be slightly longer to type but again not longer than the previous macros and to be really honest I'm not sure usage of a macro made with 90% casts (!) should be easy anyway... In fact the next commit shows this new macro is needed in only one (test) file across the entire Zephyr codebase, is that correct? If correct then this is probably not worth include/sys/util.h. Maybe some cast-free, (y-x)*sizeof() alternative local to that test file is enough.

Reply:
Please see #17518.

Reply:
Dunno what this discussion is about exactly, but I'll throw out `ptrdiff_t` just in case.